esp-hal/src/uart.rs

```diff
@@ -2245,6 +2425,20 @@ pub struct State {
 }
 
 impl Info {
+    const RX_FIFO_MAX_THRHD: u16 = if cfg!(esp32) {
```

Comment:
I think this should probably be `cfg_if`, but I actually prefer this layout, so I'll leave it up to you.

Reply:
If you prefer this, why should it be `cfg_if`?